kotest-framework/kotest-framework-plugin-gradle/src/main/kotlin/io/kotest/framework/gradle/KotestPlugin.kt

```diff
@@ -25,20 +35,37 @@ open class KotestPlugin : Plugin<Project> {
          /* ...constructionArguments = */ project
       )
 
-      // add a kotest task to all targets that have a test task
-//      project.plugins.withId("org.jetbrains.kotlin.multiplatform") {
-//         println("multiplatform=" + this::class.qualifiedName)
-//      }
-
-      // todo only add to projects which have a Test type already
-      // todo should only add to projects which have kotest on the classpath
-      project.afterEvaluate {
-         plugins.withId("org.jetbrains.kotlin.jvm") {
-            this@afterEvaluate.allprojects.forEach { subproject ->
-               subproject.tasks.register("kotest", KotestTask::class.java) {
-                  description = DESCRIPTION
-                  group = JavaBasePlugin.VERIFICATION_GROUP
-                  dependsOn(subproject.tasks.getByName(JavaPlugin.TEST_CLASSES_TASK_NAME))
+      // Configure Kotlin JVM projects
+      project.pluginManager.withPlugin(KOTLIN_JVM_PLUGIN) {
+         project.extensions.configure<KotlinJvmExtension> {
+            project.tasks.register("kotest", KotestTask::class.java) {
+               description = DESCRIPTION
+               group = JavaBasePlugin.VERIFICATION_GROUP
+               inputs.files(project.tasks.withType<KotlinCompile>().map { it.outputs.files })
+               // TODO: I think wiring dependencies,
+//               dependsOn(project.tasks.withType<KotlinCompile>())
+            }
+         }
+      }
+
+      // Configure Kotlin multiplatform projects
+      project.pluginManager.withPlugin(KOTLIN_MULTIPLATFORM_PLUGIN) {
+         project.extensions.configure<KotlinMultiplatformExtension> {
+            targets.configureEach {
```

Comment:
does targets come from the KotlinMultiplatformExtension?

Reply:
Yes :) so when I tested I created a jvm and js target, and the plugin created `kotestJvm` and `kotestJs`